core/src/main/java/nl/nn/adapterframework/doc/model/FrankElement.java

```diff
@@ -1,27 +1,225 @@
+						fullName, a.getName()));
+			} else {
+				attributeLookup.put(a.getName(), a);
+			}
+		}
+	}
+
+	public List<FrankAttribute> getAttributes(Predicate<? super FrankAttribute> filter) {
+		return attributes.stream().filter(filter).collect(Collectors.toList());
+	}
+
+	/**
+	 * Setter for config children. We prevent modifying the list of config children
+	 * because we want to maintain the private field configChildLookup.
+	 * @param children
+	 */
+	public void setConfigChildren(List<ConfigChild> children) {
+		Collections.sort(children);
+		this.configChildren = Collections.unmodifiableList(children);
+		configChildLookup = new HashMap<>();
+		for(ConfigChild c: children) {
+			ConfigChildKey key = new ConfigChildKey(c);
+			if(configChildLookup.containsKey(key)) {
+				log.warn(String.format("Different config children of Frank element [%s] have the same key", fullName));
+			} else {
+				configChildLookup.put(key, c);
+			}
+		}
+	}
+
+	public List<ConfigChild> getConfigChildren(Predicate<? super ConfigChild> filter) {
+		return configChildren.stream().filter(filter).collect(Collectors.toList());
+	}
+
+	ElementChild findElementChildMatch(ElementChild elementChild) {
+		if(elementChild instanceof FrankAttribute) {
+			return findAttributeMatch((FrankAttribute) elementChild);
+		} else if(elementChild instanceof ConfigChild) {
+			return findConfigChildMatch((ConfigChild) elementChild);
+		} else {
+			throw new IllegalArgumentException(String.format(
+					"Expected a FrankAttribute or ConfigChild, but got a [%s]",
+					elementChild.getClass().getName()));
+		}
+	}
+
+	FrankAttribute findAttributeMatch(FrankAttribute attribute) {
+		return attributeLookup.get(attribute.getName());
+	}
+
+	ConfigChild findConfigChildMatch(ConfigChild configChild) {
+		return configChildLookup.get(new ConfigChildKey(configChild));
+	}
+
+	public <T extends ElementChild> FrankElement getNextAncestor(Predicate<ElementChild> childFilter, Class<T> kind) {
+		FrankElement ancestor = parent;
+		while((ancestor != null) && (ancestor.getChildren(childFilter, kind).size() == 0)) {
+			ancestor = ancestor.getParent();
+		}
+		return ancestor;
+	}
+
+	public <T extends ElementChild> List<ElementChild> getChildren(Predicate<ElementChild> selector, Class<T> kind) {
+		List<ElementChild> result = new ArrayList<>();
+		if(kind.isAssignableFrom(FrankAttribute.class)) {
+			result.addAll(getAttributes(selector));
+		}
+		if(kind.isAssignableFrom(ConfigChild.class)) {
+			result.addAll(getConfigChildren(selector));
+		}
+		return result;
+	}
+
+	public void walkCumulativeAttributes(
+			CumulativeChildHandler<FrankAttribute> handler,
+			Predicate<ElementChild> childSelector,
+			Predicate<ElementChild> childRejector) {
+		new AncestorChildNavigation<String, FrankAttribute>(
+				handler, childSelector, childRejector, FrankAttribute.class) {
+			@Override
+			String keyOf(ElementChild child) {
+				return ((FrankAttribute) child).getName();
```

Comment:
als je getKey() in ElementChild definieert, en voor FrankAttribute implementeert als getName(), dan wordt deze code een stuk eenvoudiger.

Reply:
Gedaan.